packages/twenty-server/src/core/quick-actions/quick-actions.service.ts

```diff
@@ -0,0 +1,147 @@
+import { Injectable } from '@nestjs/common';
+
+import { v4 as uuidv4 } from 'uuid';
+
+import { Record as IRecord } from 'src/workspace/workspace-query-builder/interfaces/record.interface';
+
+import { isWorkEmail } from 'src/utils/is-work-email';
+import { stringifyWithoutKeyQuote } from 'src/workspace/workspace-query-builder/utils/stringify-without-key-quote.util';
+import { WorkspaceQueryRunnerService } from 'src/workspace/workspace-query-runner/workspace-query-runner.service';
+import { IntelligenceService } from 'src/core/quick-actions/intelligence.service';
+
+@Injectable()
+export class QuickActionsService {
+  constructor(
+    private readonly workspaceQueryRunnunerService: WorkspaceQueryRunnerService,
+    private readonly intelligenceService: IntelligenceService,
+  ) {}
+
+  async createCompanyFromPerson(id: string, workspaceId: string) {
+    const person = (
+      (await this.workspaceQueryRunnunerService.executeAndParse(
+        `query {
+        personCollection(filter: {id: {eq: "${id}"}}) {
+              edges {
+                node {
+                  id
+                  email
+                  companyId
+                }
+              }
+            }
+          }
+        `,
+        'person',
+        '',
+        workspaceId,
+      )) as IRecord
+    ).edges?.[0]?.node;
+
+    if (!person.companyId && person.email && isWorkEmail(person.email)) {
+      const companyDomainName = person.email.split('@')[1];
```

Comment:
More security accessing the array variable just in case (`?.`)
```suggestion
      const companyDomainName = person.email.split('@')?.[1];
```

Reply:
Good point thanks! I was thinking isWorkEmail would cover it, but we're never too safe 🤓